apple/MarkdownTextInputDecoratorShadowNode.mm

```diff
@@ -0,0 +1,255 @@
+  RCTMarkdownStyle *markdownStyle =
+      [[RCTMarkdownStyle alloc] initWithStruct:markdownProps.markdownStyle];
+  RCTMarkdownUtils *utils = [[RCTMarkdownUtils alloc] init];
+  [utils setMarkdownStyle:markdownStyle];
+  [utils setParserId:[NSNumber numberWithInt:markdownProps.parserId]];
+
+  // convert the attibuted string stored in state to
+  // NSAttributedString
+  auto nsAttributedString = RCTNSAttributedStringFromAttributedStringBox(
+      stateData.attributedStringBox);
+
+  auto newStateData = TextInputState(stateData);
+
+  if (stateData.attributedStringBox.getMode() ==
+      AttributedStringBox::Mode::Value) {
+
+    // Handles the first render, where the text stored in props is
+    // different than the one stored in state. The one in state is empty,
+    // while the one in props is passed from JS. If we don't update the
+    // state here, we'll end up with a one-default-line-sized text input
+    if (textInputState.getRevision() == State::initialRevisionValue) {
+      auto plainStringFromState =
+          std::string([[nsAttributedString string] UTF8String]);
+
+      if (plainStringFromState != textInputProps.text) {
+        // creates new AttributedString from props, adapted from
+        // TextInputShadowNode (ios one, text inputs are
+        // platform-specific)
+        auto attributedString = AttributedString{};
+        attributedString.appendFragment(AttributedString::Fragment{
+            textInputProps.text, defaultTextAttributes});
+
+        auto attachments = BaseTextShadowNode::Attachments{};
+        BaseTextShadowNode::buildAttributedString(
+            defaultTextAttributes, *textInput, attributedString, attachments);
+
+        // convert the newly created attributed string to
+        // NSAttributedString
+        nsAttributedString = RCTNSAttributedStringFromAttributedStringBox(
+            AttributedStringBox{attributedString});
+      }
+    }
+
+    // apply markdown
+    auto newString = [utils parseMarkdown:nsAttributedString
+                withDefaultTextAttributes:defaultNSTextAttributes];
+
+    // create a clone of the old TextInputState and update the
+    // attributed string box to point to the string with markdown
+    // applied
+    newStateData.attributedStringBox =
+        RCTAttributedStringBoxFromNSAttributedString(newString);
+  } else if (stateData.attributedStringBox.getMode() ==
+             AttributedStringBox::Mode::OpaquePointer) {
+
+    // apply markdown
+    auto newString = [utils parseMarkdown:nsAttributedString
+                withDefaultTextAttributes:defaultNSTextAttributes];
+
+    // create a clone of the old TextInputState and update the
+    // attributed string box to point to the string with markdown
+    // applied
+    newStateData.attributedStringBox =
+        RCTAttributedStringBoxFromNSAttributedString(newString);
+  }
+
+  textInput->setStateData(std::move(newStateData));
+}
+
+// this is private in YogaLayoutableShadowNode
```

Comment:
Let's add a comment that this function was copied from some file in react-native